### PR TITLE
Fix tests for tenancy enforcement

### PR DESCRIPTION
**Pull request by gtanzillo**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def with_current_request
         else
           ActsAsTenant.without_tenant { yield }
         end
-      rescue TopologicalInventory::Api::NoTenantError, KeyError
+      rescue TopologicalInventory::Api::NoTenantError, KeyError, ManageIQ::API::Common::IdentityError
```
Comment (agrare) [inline app/controllers/application_controller.rb]: Do we need our own NoTenantError anymore?  It looks like `current.user.tenant` might blow up before we get our own keyerror.

File: spec/requests/api/v0.0/authentications_spec.rb [hunk 2]
```diff
@@ -1,12 +1,14 @@
 RSpec.describe("v0.0 - Authentications") do
+  include ::Spec::Support::TenantIdentity
+
+  let(:headers)         { {"CONTENT_TYPE" => "application/json", "x-rh-identity" => identity} }
   let(:attributes)      { {"username" => "test_name", "password" => "Test Password", "tenant_id" => tenant.id.to_s, "resource_type" => "Tenant", "resource_id" => tenant.id.to_s} }
   let(:collection_path) { "/api/v0.0/authentications" }
-  let(:tenant)          { Tenant.find_or_create_by!(:name => "default", :external_tenant => "external_tenant_uuid")}
 
-  describe("/api/v0.0/authentications") do
+  pending("/api/v0.0/authentications") do
```
Comment (agrare) [inline spec/requests/api/v0.0/authentications_spec.rb]: Interesting, what fails here?

**Comment by agrare** [comment]
Nice @gtanzillo !  Fixing all those specs must have gotten repetitive

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (gtanzillo) [reply app/controllers/application_controller.rb]: I think you are right. That error was raised before we started auto creating the tenant. Let me verify that.

File: spec/requests/api/v0.0/authentications_spec.rb (hunk 2 above)
Comment (gtanzillo) [reply spec/requests/api/v0.0/authentications_spec.rb]: That was a local change that shouldn't have been pushed. I've reversed it

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (gtanzillo) [reply app/controllers/application_controller.rb]: @agrare I pushed another commit to remove `NoTenantError`

Comment (agrare) [reply app/controllers/application_controller.rb]: :+1:

**Comment by gtanzillo** [comment]
Thanks! @agrare

**Pull request merged**